### ISSUE TITLE
dotnet commands interpret string content

### ISSUE THREAD

**Issue by MangelMaxime** [NuGet/Home#6063]
Hello, I am having problem swith the pack command which seems to interpret the content of a string argument.

## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe): dotnet.exe

NuGet version (x.x.x.xxx): NuGet Command Line 4.3.0.5

dotnet.exe --version (if appropriate): 2.0.0

OS version (i.e. win10 v1607 (14393.321)): Window 10 pro

```shell
C:\Users\mange>wmic os get version
Version
10.0.15063
```

## Detailed repro steps so we can see the same problem

1. Execute the next command:

```shell
dotnet pack -c Release /p:Version=1.0.0-beta-002 /p:PackageReleaseNotes="* BREAKING CHANGE: Fix icons display and implement Font Awesome helpers
 * Fix #61: Add `is-fullwidth` support for Table element
 * Fix #59: Add color support for the navbar compoenent
 * Fix #67: Enforce DSL synthax with `[<RequireQualifiedAccess>]` attribute
 * BREAKING CHANGE: Fix #70: Make button support different elements. To have the same behavior as before, replace: `Button.button` with `Button.button_a`"
```

The command fails with:

```shell
MSBUILD : error MSB1006: La propriété n'est pas valide.
Commutateur :  replace: `Button.button` with `Button.button_a`
```
Basically it's saying: `The property is invalid`

When removing the `,` from the string the command pass.

```shell
dotnet pack -c Release /p:Version=1.0.0-beta-003 /p:PackageReleaseNotes="* BREAKING CHANGE: Fix icons display and implement Font Awesome helpers
 * Fix #61: Add `is-fullwidth` support for Table element
 * Fix #59: Add color support for the navbar component
 * Fix #67: Enforce DSL synthax with `[<RequireQualifiedAccess>]` attribute
 * BREAKING CHANGE: Fix #70: Make button support different elements. To have the same behavior as before replace `Button.button` with `Button.button_a`"
```

The command has been remove on the last line: `before, replace` --> `before replace`

Original report for this issue was: https://github.com/Microsoft/dotnet/issues/517

**Comment by rrelyea** [comment]
if this happens for non-nuget commands, this may be a "dotnet.exe" or "msbuild" issue.
Let's look at our handling first.
